### PR TITLE
[Refactor] Refactor bignum header file into several files

### LINKED ISSUE #717: Refactor bignum.h
See comments in #681

### PR DESCRIPTION
**Pull request by Warrows**
Closes #717

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -324,6 +324,14 @@ libzerocoin_libbitcoin_zerocoin_a_SOURCES = \
...
   libzerocoin/SpendType.h \
   libzerocoin/ZerocoinDefines.h \
+  libzerocoin/bignum.cpp
+if USE_NUM_GMP
+  libzerocoin_libbitcoin_zerocoin_a_SOURCES += libzerocoin/bignum_gmp.cpp
+endif
+if USE_NUM_OPENSSL
+  libzerocoin_libbitcoin_zerocoin_a_SOURCES += libzerocoin/bignum_openssl.cpp
+endif
+libzerocoin_libbitcoin_zerocoin_a_SOURCES += \
```
Comment (Fuzzbawls) [inline src/Makefile.am]: Would prefer to have these conditionals done after the initial list of sources instead of wedged in the middle to keep things looking cleaner.

File: src/Makefile.am [hunk 2]
```diff
@@ -312,9 +312,9 @@ crypto_libbitcoin_crypto_a_SOURCES = \
 libzerocoin_libbitcoin_zerocoin_a_CPPFLAGS = $(AM_CPPFLAGS) $(BOOST_CPPFLAGS)
 libzerocoin_libbitcoin_zerocoin_a_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
 libzerocoin_libbitcoin_zerocoin_a_SOURCES = \
+  libzerocoin/bignum.h \
   libzerocoin/Accumulator.h \
   libzerocoin/AccumulatorProofOfKnowledge.h \
-  libzerocoin/bignum.h \
```
Comment (Fuzzbawls) [inline src/Makefile.am]: Unnecessary change in ordering

File: src/libzerocoin/bignum_gmp.cpp [hunk 3]
```diff
@@ -0,0 +1,337 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2012 The Bitcoin developers
+// Copyright (c) 2017-2018 The PIVX developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#include "bignum.h"
```
Comment (Fuzzbawls) [inline src/libzerocoin/bignum_gmp.cpp]: Add empty line between copyright header and include

File: src/libzerocoin/bignum_openssl.cpp [hunk 4]
```diff
@@ -0,0 +1,504 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2012 The Bitcoin developers
+// Copyright (c) 2017-2018 The PIVX developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#include "bignum.h"
```
Comment (Fuzzbawls) [inline src/libzerocoin/bignum_openssl.cpp]: Add empty line between copyright header and include

**Comment by Warrows** [comment]
Styling changes done.

**Comment by furszy** [comment]
Concept ACK.

**Comment by Fuzzbawls** [comment]
needs rebase

**Comment by Warrows** [comment]
Rebased.

**Comment by furszy** [comment]
ACK `4e4975b`
